plugin/validator_test.go

```diff
@@ -398,9 +405,159 @@ func TestValidator_ValidatePathBad(t *testing.T) {
+				"config": {
+					"path": "length must be at least 1"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path not starting with /",
+			path: "path",
+			expectedErr: `{
+				"config": {
+					"path": "should start with: / (fixed path) or ~/ (regex path)"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path not starting with /",
+			path: "path/200",
+			expectedErr: `{
+				"config": {
+					"path": "should start with: / (fixed path) or ~/ (regex path)"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "/foo//bar",
+			expectedErr: `{
+				"config": {
+					"path": "must not have empty segments"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "/foo/bar//",
+			expectedErr: `{
+				"config": {
+					"path": "must not have empty segments"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "//foo/bar",
+			expectedErr: `{
+				"config": {
+					"path": "must not have empty segments"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "[[/users/|foo/profile]]",
+			expectedErr: `{
+				"config": {
+					"path": "should start with: / (fixed path) or ~/ (regex path)"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "[[/users/(this|foo/profile)]]",
+			expectedErr: `{
+				"config": {
+					"path": "should start with: / (fixed path) or ~/ (regex path)"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "/some%2words",
+			expectedErr: `{
+				"config": {
+					"path": "invalid url-encoded value: '%2w'"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "/some%0Xwords",
+			expectedErr: `{
+				"config": {
+					"path": "invalid url-encoded value: '%0X'"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "/some%2Gwords",
+			expectedErr: `{
+				"config": {
+					"path": "invalid url-encoded value: '%2G'"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "/some%20words%",
+			expectedErr: `{
+				"config": {
+					"path": "invalid url-encoded value: '%'"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "/some%20words%a",
+			expectedErr: `{
+				"config": {
+					"path": "invalid url-encoded value: '%a'"
+				}
+			}`,
+		},
+		{
+			name: "errors out on path containing not allowed elements",
+			path: "/some%20words%ax",
+			expectedErr: `{
+				"config": {
+					"path": "invalid url-encoded value: '%ax'"
+				}
+			}`,
+		},
+	}
+	for _, path := range badPaths {
+		t.Run(path.name, func(t *testing.T) {
+			err := v.Validate(fmt.Sprintf(pathTestPlugin, path.path))
+			require.Error(t, err)
+			assert.JSONEq(t, path.expectedErr, err.Error())
```

Comment:
Could we write a couple tests for this for `typedefs.router_paths`? I know this is just an array of `typedefs.router_path`, but it will allow for complete typedef coverage. The easiest would be all of the good router paths in one test, then a mixture of good and bad for failure test; WDYT?

Reply:
There are no existing tests for `typedefs.paths`, so I was just matching what was already done. With that said, yes we can update the code to support running tests for those types.